src/adapters/heads/model_mixin.py

```diff
@@ -0,0 +1,727 @@
+        """
+        Deletes the prediction head with the specified name from the model.
+
+        Args:
+            head_name (str): The name of the prediction to delete.
+        """
+        if head_name not in self.config.prediction_heads:
+            logger.info("No prediction head '%s' found for deletion. Skipping.", head_name)
+            return
+        del self.config.prediction_heads[head_name]
+        del self.heads[head_name]
+        if self.active_head == head_name:
+            self.active_head = None
+
+    def _get_used_heads(self, head_name: str = None):
+        if head_name:
+            used_heads = [head_name]
+        # together with context, check if we have heads at all to allow for models without heads
+        elif len(self.heads) > 0 and AdapterSetup.get_context_head_setup():
+            used_heads = AdapterSetup.get_context_head_setup()
+            if isinstance(used_heads, str):
+                used_heads = [used_heads]
+        elif self._active_heads:
+            used_heads = self._active_heads
+        else:
+            return []
+
+        head_modules = []
+        for head in used_heads:
+            if head not in self.heads:
+                raise ValueError("Unknown head_name '{}'".format(head))
+            head_modules.append(self.heads[head])
+
+        return head_modules
+
+    def forward_head(
+        self,
+        all_outputs,
+        head_name=None,
+        cls_output=None,
+        attention_mask=None,
+        return_dict=False,
+        context=None,
+        **kwargs
+    ):
+        """
+        The forward pass through a prediction head configuration. There are three ways to specify the used prediction
+        head configuration (in order of priority):
+
+            1. If a head_name is passed, the head with the given name is used.
+            2. If the forward call is executed within an ``AdapterSetup`` context, the head configuration is read from
+               the context.
+            3. If the ``active_head`` property is set, the head configuration is read from there.
+
+        Args:
+            all_outputs (dict): The outputs of the base model.
+            head_name (str, optional): The name of the prediction head to use. If None, the active head is used.
+            cls_output (torch.Tensor, optional): The classification output of the model.
+            attention_mask (torch.Tensor, optional): The attention mask of the model.
+            return_dict (bool): Whether or not to return a ``ModelOutput`` instead of a plain tuple.
+            get_cls_from_eos_tokens (bool):
+                If set to True, retrieve classifier token representations from the last <eos> token in the sequence.
+                Setting to True requires `eos_mask` to be passed as well.
+            **kwargs: Additional keyword arguments passed to the forward pass of the head.
+        """
+        used_head_modules = self._get_used_heads(head_name)
```

Comment:
I find this variable naming unintuitive as the head will be used. Maybe rename to `active_head_modules` or something similar?